test.rb

```diff
@@ -0,0 +1,211 @@
+# frozen_string_literal: true
+
+require 'zlib'
+require 'stackprof'
+require 'googleauth'
+require 'google/apis/cloudprofiler_v2'
+
+require './profile_pb'
+
+Cloudprofiler = Google::Apis::CloudprofilerV2
+
+PROFILE_TYPES = {
+  'CPU' => :cpu,
+  'WALL' => :wall,
+  'HEAP_ALLOC' => :object
+}.freeze
+SERVICE_REGEXP = /^[a-z]([-a-z0-9_.]{0,253}[a-z0-9])?$/.freeze
+
+SCOPES = ['https://www.googleapis.com/auth/cloud-platform'].freeze
+
+def parse_duration(d)
+  m = /^(\d+)s$/.match(d)
+  return 10 if m.nil?
+
+  Integer(m[1])
+end
+
+class Profiler
+  def initialize(service:, project_id:, serviceVersion: nil, debug_logging: false, instance: nil, zone: nil)
+    raise "service must match #{SERVICE_REGEXP}" unless SERVICE_REGEXP =~ service
```

Comment:
This would normally be an ArgumentError rather than a RuntimeError
```suggestion
    raise ArgumentError, "service must match #{SERVICE_REGEXP}" unless SERVICE_REGEXP =~ service
```